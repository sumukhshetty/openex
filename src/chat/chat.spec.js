// for these tests to work you need to disable react-ga on index.js line 53 and logpageView on line 96 of index.js

// I imported teh action here so as to not have to disable redner and firebase extenions in teh imported file.

const addMessage = (key, message, tradeId) => {
  return {
    type: 'ADD_MESSAGE',
    content: message.content,
    key,
    timeStamp: message.timeStamp,
    uid: message.uid,
    tradeId
  }
}

test('addMessage', () => {
  expect(
    addMessage(
      11,
      {
        type: 'ADD_MESSAGE',
        content: 'testing addMessage action',
        timeStamp: 11223344556677,
        uid: 9876
      },
      1234
    )
  ).toMatchSnapshot()
})

const createMessage = ({
  content,
  uid,
  tradeId,
  download,
  purchaseRequest
}) => {
  return dispatch => {
    const message = {
      content,
      uid,
      timeStamp: Date.now(),
      download
    }
  }
}

test('createMessage', () => {
  expect(
    addMessage(
      11,
      {
        type: 'ADD_MESSAGE',
        content: 'testing addMessage action',
        timeStamp: 11223344556677,
        uid: 9876
      },
      1234
    )
  ).toMatchSnapshot()
})
