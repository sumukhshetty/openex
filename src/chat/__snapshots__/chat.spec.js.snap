// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMessage 1`] = `
Object {
  "content": "testing addMessage action",
  "key": 11,
  "timeStamp": 11223344556677,
  "tradeId": 1234,
  "type": "ADD_MESSAGE",
  "uid": 9876,
}
`;

exports[`createMessage 1`] = `
Object {
  "content": "testing addMessage action",
  "key": 11,
  "timeStamp": 11223344556677,
  "tradeId": 1234,
  "type": "ADD_MESSAGE",
  "uid": 9876,
}
`;
